### ISSUE TITLE
Using two sequences in the same table

### ISSUE THREAD

**Issue by sc0rp10** [doctrine/doctrine2#6422]
I need to handle two sequences for the same table like this:
```php
class Foo
{
    /**
     * @ORM\Id
     * @ORM\Column(type="integer")
     * @ORM\GeneratedValue(strategy="SEQUENCE")
     * @SequenceGenerator(sequenceName="a_seq")
     */
    private $a;

    /**
     * @ORM\Column(type="integer")
     * @ORM\GeneratedValue(strategy="SEQUENCE")
     * @SequenceGenerator(sequenceName="b_seq")
     */
    private $b;
}
```
But insert has failed with `not null violation` on "b".
A query looks like `INSERT INTO foo (a, b) VALUES (42, null)`.
My DDL allow using`DEFAULT nextval('b_seq'::regclass);`, but Doctrine pass `null` explicit and breaks query.
How can I avoid them?

**Comment by Ocramius** [comment]
This is not supported, as the ORM will only read a single insert identifier each time. It's a known limitation, and it also isn't possible to mix auto-incremental/sequence fields with assigned fields at the moment.

**Comment by sc0rp10** [comment]
Maybe is there some hack with metadata or another 'hacky' way to use them?

**Comment by Ocramius** [comment]
@sc0rp10 no, the limitation is [in the persisters](https://github.com/doctrine/doctrine2/blob/7bb02d0dbdc337f98fa94078985baa713c24bb65/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php#L286-L293)

**Comment by Majkl578** [comment]
This should be possible in develop (maybe not just yet, but the code is ready for this) -- thus adding to 3.0 backlog and removing wontfix, and we shall see how this goes. :)
